TLSharp.Core/Auth/Authenticator.cs

```diff
@@ -1,35 +1,38 @@
-﻿using System.Threading.Tasks;
+﻿using System.Threading;
+using System.Threading.Tasks;
 using TLSharp.Core.Network;
 
 namespace TLSharp.Core.Auth
 {
     public static class Authenticator
     {
-        public static async Task<Step3_Response> DoAuthentication(TcpTransport transport)
+        public static async Task<Step3_Response> DoAuthentication(TcpTransport transport, CancellationToken token)
```

Comment:
can you please make CancellationToken parameters be optional, and default to the value `CancellationToken.None` please?